### PR TITLE
Add recipe modin

### PR DESCRIPTION
**Pull request by devin-petersohn**
cc @h-vetinari 

Derived and rebased from #8471. 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by devin-petersohn** [comment]
@h-vetinari I updated the source and version, reset to PyPI because that seems to be working now. I also squashed to a single commit after all tests pass. Please verify that you are still willing to be a recipe maintainer for review purposes.

File: recipes/modin/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  version: "{{ version }}"
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: cb11d4f62453d47f0d014e2fd9b90fd001c6738ae377e929ae1996d860767c30 
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: true  # [py<35]
```
Comment (synapticarbors) [inline recipes/modin/meta.yaml:15]: I would suggest removing the `skip` line and make this package `noarch: python`, and then change `python` to `python >=3.5` under `host` and `run`

File: recipes/modin/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (synapticarbors) [inline recipes/modin/LICENSE:1]: A license header is included in every source file in the package, so I'm not sure about whether we need to manually package the license file. Then the question would be how to specify the `license_file` below. I'm not sure best how to handle this.

File: recipes/modin/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - python
+  run:
+    - python
+    - pandas ==1.0.3
+    - dask >=2.1.0
+    - distributed >=2.3.2
+
+test:
+  imports:
+    - modin
```
Comment (synapticarbors) [inline recipes/modin/meta.yaml:29]: Also maybe do some imports of submodules. e.g.:

```
- modin.pandas
- modin.sql
- modin.data_management.functions
...
```

**Comment by devin-petersohn** [comment]
New Tests pass, version string formatted correctly, simple dataframe created correctly.

@synapticarbors Please let me know if there is some additional change needed for the license file.

**Pull request merged**